base/essentials.jl

```diff
@@ -712,6 +712,33 @@ function invokelatest(@nospecialize(f), @nospecialize args...; kwargs...)
     Core._apply_latest(inner)
 end
 
+"""
+   invokefrozen(f, rt, args...; kwargs...)
+
+Calls `f(args...;kwargs...)` but allows the method of `f` at its current age to
+be directly executed without a world-age error. This is useful in specialized
+circumstances, such as code generation, where post-265 automatic function
+updating behavior may be non-sensical and impede usage. This has a very
+small overhead compared to calling `f` directly, and is much faster than
+`invokelatest`.
+"""
+@inline function invokefrozen(f, rt, args...; kwargs...)
+    if isempty(kwargs)
```

Comment:
Seems to add a lot more overhead when then are `kwargs`:
```julia
julia> f(;kwargs...) = kwargs[:a] + kwargs[:b]
f (generic function with 1 method)

julia> @btime f(;a=3,b=4)
  1.152 ns (0 allocations: 0 bytes)
7

julia> @btime invokefrozen(f,Int;a=3,b=4)
  195.901 ns (4 allocations: 128 bytes)
7
```

I tried splitting `invokefrozen(f, rt, args...)` into a separate method to avoid the runtime `kwargs` check, but somehow this makes it slower